### PR TITLE
doc updates, platform check, README fix

### PR DESCRIPTION
**Pull request by oyvindronningstad**
Will be squashed

Mention nRF platforms in various docs.

Add CMake check to ensure no-one does e.g. -DTFM_PLATFORM=nordic_nrf/nrf9160
(should be -DTFM_PLATFORM=nordic_nrf/nrf9160dk_nrf9160)

Fix duplicate line in READMEs

Signed-off-by: Øyvind Rønningstad <oyvind.ronningstad@nordicsemi.no>

### CODE REVIEW

**Comment by oyvindronningstad** [comment]
cc @ioannisg @anangl

File: docs/getting_started/tfm_secure_boot.rst [hunk 1]
```diff
@@ -254,6 +254,10 @@ modes are supported by which platforms:
 +---------------------+-----------------+---------------+----------+----------------+--------------+
 | STM_NUCLEO_L552ZE_Q | No              | Yes           | No       | No             | No           |
 +---------------------+-----------------+---------------+----------+----------------+--------------+
+| nRF5340             | No              | Yes           | No       | No             | No           |
```
Comment (ioannisg) [inline docs/getting_started/tfm_secure_boot.rst]: I think these are "DKs" not CPUs, so I would add PDK and DK, respectively.
E.g. NUCLEO is a board name, for STM32.

File: docs/getting_started/tfm_integration_guide.rst [hunk 2]
```diff
@@ -29,6 +29,8 @@ The hardware platforms currently supported are:
 - CoreLink SSE-200 Subsystem for MPS3 (AN524)
 - DesignStart FPGA on Cloud: Cortex-M33 based platform (SSE-200_AWS)
 - STM32L5xx: Cortex-M33 based platform (STM32L562 and STM32L552 socs)
+- nRF9160 development kit (Cortex-M33)
```
Comment (ioannisg) [inline docs/getting_started/tfm_integration_guide.rst]: Could you align with the names used in upstream Zephyr ?

File: platform/ext/target/nordic_nrf/nrf5340pdk_nrf5340_cpuapp/README.rst [hunk 3]
```diff
@@ -36,7 +36,6 @@ following commands:
 
 .. code:: bash
 
-    $ mkdir build && cd build
```
Comment (ioannisg) [inline platform/ext/target/nordic_nrf/nrf5340pdk_nrf5340_cpuapp/README.rst:39]: thanks

File: docs/getting_started/tfm_integration_guide.rst (hunk 2 above)
Comment (oyvindronningstad) [reply docs/getting_started/tfm_integration_guide.rst]: Like so?
```suggestion
- nrf9160dk_nrf9160 (Cortex-M33)
```

Comment (anangl) [reply docs/getting_started/tfm_integration_guide.rst]: I think it should be "nRF9160 DK". That's what is used in Nordic infocenter and [website](https://www.nordicsemi.com/Software-and-Tools/Development-Kits/nRF9160-DK), and also in Zephyr's [documentation](https://docs.zephyrproject.org/latest/boards/arm/nrf9160dk_nrf9160/doc/index.html).

Comment (ioannisg) [reply docs/getting_started/tfm_integration_guide.rst]: agree

File: docs/getting_started/tfm_integration_guide.rst [hunk 4]
```diff
@@ -29,6 +29,8 @@ The hardware platforms currently supported are:
 - CoreLink SSE-200 Subsystem for MPS3 (AN524)
 - DesignStart FPGA on Cloud: Cortex-M33 based platform (SSE-200_AWS)
 - STM32L5xx: Cortex-M33 based platform (STM32L562 and STM32L552 socs)
+- nrf9160dk_nrf9160 development kit (Cortex-M33)
```
Comment (ioannisg) [inline docs/getting_started/tfm_integration_guide.rst]: I would use the same names as in the tfm_secure_boot.rst, below

**Review by anangl — APPROVED** [review]

File: docs/getting_started/tfm_integration_guide.rst [hunk 5]
```diff
@@ -29,6 +29,8 @@ The hardware platforms currently supported are:
 - CoreLink SSE-200 Subsystem for MPS3 (AN524)
 - DesignStart FPGA on Cloud: Cortex-M33 based platform (SSE-200_AWS)
 - STM32L5xx: Cortex-M33 based platform (STM32L562 and STM32L552 socs)
+- nRF9160 DK (Cortex-M33)
+- nRF5340 PDK (Cortex-M33 application core)
```
Comment (ioannisg) [inline docs/getting_started/tfm_integration_guide.rst]: nit: Application MCU: this is the term i used in the READMEs

**Review by ioannisg — APPROVED** [review]
just a minor comment about the nRF530 Application MCU

**Pull request merged**